packages/grid/src/mouse-handlers/GridLinkMouseHandler.ts

```diff
@@ -0,0 +1,267 @@
+          // Check if the x position is less than the grid x, then linkWidth should be shifted by gridX - startX
+          linkWidth -= startX < gridX ? gridX - startX : 0;
+
+          const left = Math.max(gridX, gridX + startX);
+          const right = clamp(
+            left + linkWidth,
+            left,
+            gridWidth - verticalBarWidth
+          );
+
+          const cursorIsInBounds =
+            x >= left && x <= right && y >= top && y <= bottom;
+
+          if (cursorIsInBounds) {
+            // Reset font
+            context.restore();
+
+            this.currentLinkBox = {
+              left,
+              top,
+              right,
+              bottom,
+              width: linkWidth,
+              height: linkHeight,
+              value,
+              href,
+            };
+
+            return this.currentLinkBox;
+          }
+        }
+      }
+    }
+
+    // There are no links or the mouse is not on a link, restore context and currentLinkBox
+    context.restore();
+    this.currentLinkBox = undefined;
+
+    return null;
+  }
+
+  /**
+   * Set the cursor to pointer when hovering of a link
+   * @param gridPoint Mouse position information
+   * @param grid The grid
+   * @returns False
+   */
+  private setCursor(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    if (GridLinkMouseHandler.isHoveringLink(gridPoint, grid) != null) {
+      this.cursor = 'pointer';
+    } else {
+      this.cursor = null;
+    }
+    return false;
+  }
+
+  onMove(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    return this.setCursor(gridPoint, grid);
+  }
+
+  onDown(
+    gridPoint: GridPoint,
+    grid: Grid,
+    event: GridMouseEvent
+  ): EventHandlerResult {
+    this.isDown = true;
+    const link = GridLinkMouseHandler.isHoveringLink(gridPoint, grid);
+
+    // If a modifier key or shift is down, we don't want to handle it and set isDown to false so onUp won't handle it either
+    if (GridUtils.isModifierKeyDown(event) || event.shiftKey || link == null) {
+      this.isDown = false;
+      return false;
+    }
+
+    if (this.timeoutId) {
+      clearTimeout(this.timeoutId);
+    }
+    this.isHold = false;
+    this.timeoutId = setTimeout(() => {
+      this.isHold = true;
+      const { column, row } = gridPoint;
+      const { model } = grid.props;
+
+      // After 1 second, select the row and if it is an input table, start editing
+      grid.clearSelectedRanges();
+      grid.focus();
+      grid.moveCursorToPosition(column, row);
+      if (
+        isEditableGridModel(model) &&
+        column != null &&
+        row != null &&
+        model.isEditableRange(GridRange.makeCell(column, row))
+      ) {
+        grid.startEditing(column, row);
+      }
+    }, 1000);
+
+    return true;
+  }
+
+  onUp(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    const link = GridLinkMouseHandler.isHoveringLink(gridPoint, grid);
+    if (link == null || !this.isDown) {
```

Comment:
Add `|| this.isHold` here and then you don't need to check below

Unless there's a reason after a hold that the event should stop propagation (the return true)

Reply:
I think after holding, we should return true because then the event is handled (either the row is selected or is now editable) but do you think it should be false?

Reply:
I'd have to check the behavior if it's not prevented from bubbling. Should probably be fine as is